autometa/taxonomy/ncbi.py

```diff
@@ -364,27 +332,66 @@ def parse_merged(self):
-    def is_common_ancestor(self, parent_taxid, child_taxid):
-        """Determines whether the provided taxids have a non-root common ancestor
+    def is_common_ancestor(self, taxid1, taxid2):
+        """
+        Determines whether the provided taxids have a non-root common ancestor
 
         Parameters
         ----------
-        parent_taxid : int
-            Description of parameter `parent_taxid`.
-        child_taxid : int
-            Description of parameter `child_taxid`.
+        taxid1 : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+        taxid2 : int
+            identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         boolean
-            True if taxids share a common ancestor and False otherwise
+            True if taxids share a common ancestor else False
         """
-        ancestor_taxid = child_taxid
-        while ancestor_taxid != 1:
-            if parent_taxid == ancestor_taxid:
-                return True
-            ancestor_taxid = self.parent(ancestor_taxid)
-        return False
+        taxid1_lineage = self.lineage(taxid1)
+        taxid2_lineage = self.lineage(taxid2)
+        taxid1_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid1_lineage])
+        taxid2_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid2_lineage])
+        print(taxid1_lineage_taxids)
+        print(taxid2_lineage_taxids)
+        common_ancestor = taxid2_lineage_taxids & taxid1_lineage_taxids
+        print(common_ancestor)
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given taxid is a positive interger or not and is present in either 
+        'names.dmp' or 'merged.dmp'
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+
+        Returns
+        -------
+        boolean 
+            True if the 'taxid' is a positive interger and is present in either 'names.dmp' or 'merged.dmp' else False 
+        """
+        try:
+            taxid = float(taxid)
+        except ValueError:
+            logger.error(
+                f'Taxid must be a positive integer! Given: {taxid}')
+            return False
+        # `is_integer()` checks if it is something like 12.0 vs. 12.9. `is_integer` returns error if only '12' is used, float(taxid) is needed
+        if not taxid.is_integer() or taxid < 0:
+            logger.error(
+                f'Taxid must be a positive integer! Given: {taxid}')
+            return False
+        # float(taxid) also comes useful here, as '562' would not be found in the database, but float(562) would
+        if not self.names.get(taxid) and not self.merged.get(taxid):
+            logger.error(
+                'Taxid not found in names.dmp and merged.dmp')
+            return False
```

Comment:
If these are not raising errors then you could log warnings. Although I'm not entirely convinced this is useful. This sounds like it is just going to clutter the logging stream

Reply:
Should I raise the error then?

Reply:
You should consider where you are applying this method within other methods in the NCBI class, then determine how you should handle this.